automation/roles/upgrade/tasks/post_upgrade.yml

```diff
@@ -139,13 +138,16 @@
       loop_control:
         index_var: idx
         label: "pg{{ idx + 1 }}-path={{ pg_new_datadir }}"
-      when: pg_path_count.stdout | int > 0
+      when:
+        - pg_path_count.stdout | int > 0
+        - "'pgbackrest' in group_names"
 
     - name: pgbackrest | Upgrade stanza "{{ pgbackrest_stanza }}"
-      delegate_to: "{{ groups['pgbackrest'][0] }}"
       run_once: true
       ansible.builtin.command: "pgbackrest --stanza={{ pgbackrest_stanza }} --no-online stanza-upgrade"
-      when: pg_path_count.stdout | int > 0 and pgbackrest_stanza_upgrade | bool
+      when:
+        - pg_path_count.stdout | int > 0 and pgbackrest_stanza_upgrade | bool
+        - "'pgbackrest' in group_names"
```

Comment:
this condition will not work because only the "primary:secondary" groups is declared in PLAY.
```yaml
- name: "(6/6) POST-UPGRADE: Analyze a PostgreSQL database (update optimizer statistics) and Post-Upgrade tasks"
  hosts: "primary:secondary"
  ...

    - name: Running Post-Upgrade tasks
      ansible.builtin.include_role:
        name: vitabaks.autobase.upgrade
        tasks_from: post_upgrade
```
For this reason, delegate_to was used.